kong/plugins/session/storage/kong.lua

```diff
@@ -38,15 +38,45 @@ local function load_session_from_cache(key)
   return kong.cache:get(cache_key, nil, load_session_from_db, key)
 end
 
+local function insert_session_metadata(metadata, session)
+  if not metadata then
+    return
+  end
+
+  local audiences = metadata.audiences
+  local subjects  = metadata.subjects
+  local count     = #audiences
+  for i = 1, count do
+    local _, err = kong.db.session_metadatas:insert({
+      sid      = session.session_id,
+      audience = audiences[i],
+      subject  = subjects[i],
+      session  = session,
+    })
+    if err then
+      kong.db.sessions:delete(session.id)
```

Comment:
Shall we assert the `err` is about the `foreign key` constraints?

Reply:
Whenever there's an error, it needs to be rolled back, no matter what kind of error, right?

Reply:
> Whenever there's an error, it needs to be rolled back, no matter what kind of error, right?

Makes sense. Is the transaction a better way to implement this feature?

Reply:
Since our DAO doesn't support transactions, this approach can ensure the consistency of `sessions` of `session_metadatas`, so I will close this conversation.